code/game/objects/structures/flora.dm

```diff
@@ -266,23 +266,40 @@
 	throwforce = 13
 	throw_speed = 2
 	throw_range = 4
+	/// Method to track plant overlay on mob for later removal
+	var/mutable_appearance/mob_overlay
 
-/obj/item/twohanded/required/kirbyplants/New()
-	..()
+/obj/item/twohanded/required/kirbyplants/Initialize(mapload)
+	. = ..()
 	icon_state = "plant-[rand(1,35)]"
 	if(prob(1))
 		icon_state = "plant-36"
 
+/obj/item/twohanded/required/kirbyplants/Destroy()
+	if(isliving(loc))
+		unhide_user(loc)
+
+	QDEL_NULL(mob_overlay)
+	return ..()
+
 /obj/item/twohanded/required/kirbyplants/equipped(mob/living/user)
 	. = ..()
 	if(wielded)
-		var/image/I = image(icon, user, icon_state)
-		I.override = TRUE
-		user.add_alt_appearance("sneaking_mission", I, GLOB.player_list)
+		hide_user(user)
+
+/obj/item/twohanded/required/kirbyplants/proc/hide_user(mob/living/user)
+	mob_overlay = mutable_appearance(icon, icon_state, user.layer, user.plane, 255, appearance_flags = RESET_COLOR | RESET_TRANSFORM | RESET_ALPHA | KEEP_APART)
+	user.add_overlay(mob_overlay)
+	user.alpha = 0
+
+/obj/item/twohanded/required/kirbyplants/proc/unhide_user(mob/living/user)
+	user.cut_overlay(mob_overlay)
+	user.alpha = initial(user.alpha)
+	QDEL_NULL(mob_overlay)
 
 /obj/item/twohanded/required/kirbyplants/dropped(mob/living/user)
 	..()
-	user.remove_alt_appearance("sneaking_mission")
+	unhide_user(user)
```

Comment:
Is it possible that this might not always be called (say, on a forceMove)?

Reply:
A forcemove should bring the plant pot with you, I'm pretty sure. Since it's an item in your hands. I'll do some testing soon.

Reply:
Oh, I meant a forcemove of the plant pot itself. Like a summon item spell or something.

Reply:
![image](https://user-images.githubusercontent.com/12197162/235741718-26c9a35c-9cf9-4f20-944f-b8954c6096b3.png)

So uh, directly calling forcemove doesn't even remove it from your inventory, so if that happens there are some bigger issues to worry about. I'll test summon now but I'd imagine it has some sort of safety involving dropping first.

Reply:
Summoning and such works as intended, even instantly hides you, pretty poggers.

Marking as resolved. If you can think of more edge cases to run by me, please let me know.